core/src/main/scala/scalaz/Foldable.scala

```diff
@@ -329,12 +329,13 @@ trait Foldable[F[_]]  { self =>
    * Splits the elements into groups that alternatively satisfy and don't satisfy the predicate p.
    */
   def splitWith[A](fa: F[A])(p: A => Boolean): List[NonEmptyList[A]] =
-    foldRight(fa, (List[NonEmptyList[A]](), None : Option[Boolean]))((a, b) => {
+    foldRight(fa, (List.empty[NonEmptyList[A]], Maybe.empty[Boolean]))((a, b) => {
       val pa = p(a)
       (b match {
-        case (_, None) => NonEmptyList(a) :: Nil
-        case (x, Some(q)) => if (pa == q) (a <:: x.head) :: x.tail else NonEmptyList(a) :: x
-      }, Some(pa))
+        case (_, Maybe.Empty()) => List(NonEmptyList(a))
+        case (Nil, _) => List(NonEmptyList(a))
```

Comment:
This is an unreachable branch, right? Let's make illegal state unrepresentable by folding to `Maybe[(NonEmptyList[NonEmptyList[A]], Boolean)]` (instead of `(List[NonEmptyList[A]], Maybe[Boolean])`).

Reply:
Excellent! will do.